### ISSUE TITLE
Bug: Airtable Upsert Incorrectly Marks Linked Fields as Read Only

### ISSUE THREAD

**Issue by dkindlund** [n8n-io/n8n#8617]
### Bug Description

When adding a new Airtable Upsert node to a workflow, after letting the node update with the right table and fields, it seems that the node *always* marks fields that contain 1 or more linked records to be *readonly*.  This is a problem because I want to be able to use the *typecast* feature to specify a string upon upsert into this field, where Airtable would automatically lookup and link to the already created record (if it already exists in the linked table).

This is a huge quality of life issue, because the workaround requires me to:
* copy the Airtable Upsert node OUT of the n8n workflow
* in a text editor, find the (incorrectly marked) field that's labeled `readonly: true` and set it to be `readonly: false`
* then paste the node BACK into the n8n workflow

This gets pretty bad when you're juggling more than one airtable table in a given workspace.

### To Reproduce

1. Create a new workflow in n8n
2. Define new Airtable credentials that can read/write to a table
3. Create a new Airtable table with one field that points to another linked table
4. Add the Airtable Upsert node to your n8n workflow
5. Specify the Airtable credentials and table you wish to upsert
6. See that the field containing the linked records is marked `readonly: true`

### Expected behavior

I expect all Airtable fields in the upsert node that map to linked records to have `readonly: false`.

### Operating System

Google Cloud Run

### n8n Version

1.27.2

### Node.js Version

18.10

### Database

PostgreSQL

### Execution mode

main (default)

**Comment by Joffcom** [comment]
Hey @dkindlund,

Thanks for the report, I don't believe this is a bug and it was an intentional choice but I will need to check the other report we had of this to see what we planned to do.

**Comment by Joffcom** [comment]
Found it, This is in our backlog to change and has an internal dev id of `NODE-981`

**Comment by dkindlund** [comment]
Okay, this bug is a pretty big blocker for anyone looking to built Airtable automations on top of n8n.  Otherwise, you're forcing your users to manually hack through raw JSONs for each upsert node.

**Comment by Joffcom** [comment]
We do have this down to be changed in the future, it looks like this is only the second time this has come up which is why it has not already been changed.

**Comment by dkindlund** [comment]
@Joffcom , understood.  Considering you're trying to compete with [make.com](https://n8n.io/vs/make/), just as an FYI, Airtable is hugely popular with legacy make.com customers.  [You can even see for yourself.](https://www.google.com/search?q=site%3Acommunity.make.com+airtable)

My point is that if you're looking to foster more adoption/migration from make.com to n8n.io, having first-class support for all of basic Airtable operations would greatly move the needle.  I think Airtable is like one of their [top 5 most popular integrations](https://www.google.com/search?q=make.com+airtable).

**Comment by Joffcom** [comment]
Hey @dkindlund,

It is also popular with our users and we will change this however this is still only the second time this specific issue has come up, when it comes to prioritising issues we base part of it on how many users are impacted and how often it has come up as an issue.

This will be changed it is just right now we have higher priority issues to fix, I don't think it will be that long for this to happen I just don't have a timeline for it.

**Comment by dkindlund** [comment]
Hey @Joffcom , it seems I've also been bitten by this other known issue with Airtable:
https://community.n8n.io/t/airtable-typecast-is-not-working/30745/12

^ Is there already a ticket tracking that one as well?  If not, let me know and I'll open a new issue to track it.

The main issue there is:
* If you create a Airtable table with a single select field that has like 3 enumerated types (options)
* Then you proceed to Upsert an Airtable record with `typecast` set to `true` but the single select field you specify isn't one of the existing 3 enumerated types (but let's say it's a 4th type)
* Then n8n will throw an error, saying it can't Upsert with the new 4th type

The workaround is to:
* Copy the Airtable Upsert node out of the editor
* Paste the raw JSON into a local text editor
* Change the `type` field from `options` to `string` for the corresponding single select field (and remove the `options` field)
* At that point, the workflow will function correctly

It seems like the pattern to solving most of the Airtable limitations is this JSON export/fiddle/paste thing that most people won't really understand or get.

**Comment by Joffcom** [comment]
Hey @dkindlund

That issue does not look to be a bug to me, If you have set a single field type in Airtable using an enum and you try to add an option that isn't there I would expect it to throw an error as it doesn't know about the 4th type.

In this case Airtable is responding to let you know this, More importantly the `typecast` option is not really related to this and when set it just tells Airtable to treat data in a different way... We don't change anything it is put on Airtable to handle that.

From the Airtable API Docs

> The Airtable API will perform best-effort automatic data conversion from string values if the typecast parameter is passed in. Automatic conversion is disabled by default to ensure data integrity, but it may be helpful for integrating with 3rd party data sources.

Sounds like we would need a feature request to allow overwriting the select field.

**Comment by marvinwu** [comment]
> Okay, this bug is a pretty big blocker for anyone looking to built Airtable automations on top of n8n. Otherwise, you're forcing your users to manually hack through raw JSONs for each upsert node.

I agree with that assessment., there are [other reports](https://community.n8n.io/t/not-able-to-link-records-on-airtable-in-n8n-io/35471/7) on this issue in the n8n community forum:

Airtable + n8n is a powerful combination, and updating linked records is a very common use case. It probably deserves a higher priority, especially if the fix is straightforward.

**Comment by tim1smith** [comment]
the same problem, but i use create record & update record airtable nodes . Typecast isn't working. Can't create new value in single select and multiselect.
